src/BinDeps.jl

```diff
@@ -561,6 +561,21 @@ function eval_anon_module(context, file)
     return
 end
 
+"""
+    glibc_version()
+
+For Linux-based systems, return the version of glibc in use. For non-glibc Linux and
+other platforms, returns `nothing`.
+"""
+function glibc_version()
+    Compat.Sys.islinux() || return
+    libc = ccall(:jl_dlopen, Ptr{Void}, (Ptr{Void}, UInt32), C_NULL, 0)
+    ptr = Libdl.dlsym_e(libc, :gnu_get_libc_version)
+    ptr == C_NULL && return # non-glibc
+    v = unsafe_string(ccall(ptr, Ptr{UInt8}, ()))
+    ismatch(Base.VERSION_REGEX, v) ? VersionNumber(v) : nothing
```

Comment:
I wonder if we should print out a warning here?  What should we do if we encounter a `glibc` that we can't get the version of?

Reply:
That's a good question. Doing it this way was more of a precaution; I think `gnu_get_libc_version` is always supposed to produce a valid version number, so maybe an error would be best?

Reply:
What do you think of this, @yuyichao?